src/core/aws-net-websocket-registry.adb

```diff
@@ -342,16 +364,19 @@ package body AWS.Net.WebSocket.Registry is
    begin
       Handle_Message : loop
          declare
+            WS_Id     : UID;
             WebSocket : Object_Class;
             Message   : Unbounded_String;
          begin
             Message := Null_Unbounded_String;
 
-            Message_Queue.Get (WebSocket);
+            Message_Queue.Get (WS_Id);
+
+            DB.Get_Socket (WS_Id, WebSocket);
 
             --  A WebSocket is null when termination is requested
 
-            exit Handle_Message when WebSocket = null;
+            exit Handle_Message when WS_Id = No_UID or else WebSocket = null;
```

Comment:
when WS_Id is No_UID, Get_Socket will already return null, right ? So the extra test is redundant. Or else we should check for No_UID before we even call the protected procedure Get_Socket, for efficiency